akka-persistence-query/src/main/scala/akka/persistence/query/Offset.scala

```diff
@@ -44,7 +44,7 @@ final case class TimeBasedUUID(value: UUID) extends Offset with Ordered[TimeBase
     throw new IllegalArgumentException("UUID " + value + " is not a time-based UUID")
   }
 
-  override def compare(other: TimeBasedUUID): Int = value.compareTo(other.value)
+  override def compare(other: TimeBasedUUID): Int = value.timestamp().compareTo(other.value.timestamp())
```

Comment:
We have a more comprehensive implementation in akka-persistence-cassandra. We could bring that into Akka itself instead. https://github.com/akka/akka-persistence-cassandra/blob/master/core/src/main/scala/akka/persistence/cassandra/query/UUIDComparator.scala

The UUIDComparator could be located in in `akka.util` in akka-actor since it's general purpose.

Reply:
For what I see in that `UUIDComparator`, the only thing needed is:

```scala
  private def compareULongs(l1: Long, l2: Long): Int = {
    val diff = compareUInts((l1 >> 32).toInt, (l2 >> 32).toInt)
    if (diff == 0)
      compareUInts(l1.toInt, l2.toInt)
    else
      diff
  }
```

And then the compare can be implemented as:
```scala
override def compare(other: TimeBasedUUID): Int = compareULongs(value.timestamp(), other.value.timestamp())
```

The rest it is not necessary here, since there is this [constructor check](https://github.com/akka/akka/blob/master/akka-persistence-query/src/main/scala/akka/persistence/query/Offset.scala#L44)

Reply:
That is true for, but might be good to include the full thing since we already have it. I can bring that into Akka if you don't want to spend time on it.

Reply:
I've made the requested change.